lib/dradis/plugins/mapping.rb

```diff
@@ -0,0 +1,60 @@
+
+# When you call provides :upload in your Engine, this module gets included.
+module Dradis::Plugins::Mapping
+  extend ActiveSupport::Concern
+
+  included do
+    module_parent.extend ClassMethods
+  end
+
+  module ClassMethods
+
+    def component
+      meta[:name].to_s
+    end
+
+    def field_names(source:, destination: nil, field_type: 'destination')
+      mappings = Mapping.includes(:mapping_fields).where(component: component)
+      mappings = mappings.where(source: source) if source
+      mappings = mappings.where(destination: destination) if destination
+
+      fields = mappings.pluck("mapping_fields.#{field_type}_field").uniq
+
+      if fields.empty? && source
+        default_mapping(source).keys
+      else
+        fields
+      end
+    end
+
+    def default_mapping(source)
+      self::Mapping::DEFAULT_MAPPING[source.to_sym]
+    end
+
+    def mapping(source:, destination:)
```

Comment:
wouldn't it be this method task to First search the DB and failing, return the default mapping? I.e. always returning a mapping. Perhaps that's not what we want?

We're imposing that on the caller (it seems). Are there situations in which the caller wants the DB mapping only, and wouldn't immediately request a default mapping if non exist?

Reply:
Yeah that makes a lot more sense, I don't think there are any cases where we don't want the default if there is no matching mapping. Updated 👍